mmdet3d/models/detectors/centerpoint_two_stage.py

```diff
@@ -0,0 +1,168 @@
+        self.voxel_encoder = builder.build_voxel_encoder(voxel_encoder)
+        self.middle_encoder = builder.build_middle_encoder(middle_encoder)
+
+    def extract_feat(self, points):
+        """Extract features from images and points.
+        Args:
+            points (list[torch.Tensor]): Points of each sample. The outer
+                list indicates point cloud in a batch. The shape of inside
+                point cloud is [N, C]
+
+        Returns:
+            list[torch.Tensor]: Multi-level feature maps. The shape of each
+                feature map is [B, C_i, H_i, W_i]
+        """
+        voxels, num_points, coors = self.voxelize(points)
+        voxel_features = self.voxel_encoder(voxels, num_points, coors)
+        batch_size = coors[-1, 0] + 1
+        x = self.middle_encoder(voxel_features, coors, batch_size)
+        x = self.backbone(x)
+        if self.with_neck:
+            x = self.neck(x)
+        return x
+
+    def forward_train(self, points, img_metas, gt_bboxes_3d, gt_labels_3d):
+        """Training forward function of CenterPointTwoStage.
+
+        Args:
+            points (list[torch.Tensor]): Points of each sample. The outer
+                list indicates point cloud in a batch. The shape of inside
+                point cloud is [N, C]
+            imput_metas (list[dict]): Meta info of each input.
+            gt_bboxes_3d (list[:obj:`BaseInstance3DBoxes`]):
+                GT bboxes of each sample. The bboxes are encapsulated
+                by 3D box structures.
+            gt_labels_3d (list[LongTensor]): GT labels of each sample.
+
+        Returns:
+            dict: Losses from CenterPoint Two-Stage head.
+                - loss_bbox (torch.Tensor): Loss of bboxes.
+        """
+        losses = dict()
+        # - extract feature
+        bev_feature = self.extract_feat(points)  # list[torch.Tensor]
+        # - one-stage loss
+        one_stage_head_outs = self.rpn_head(bev_feature)
+        ont_stage_loss_inputs = [
+            gt_bboxes_3d, gt_labels_3d, one_stage_head_outs
+        ]
+        one_stage_losses = self.rpn_head.loss(*ont_stage_loss_inputs)
+        losses.update(one_stage_losses)
+        # - get one-stage rois
+        rois = self.rpn_head.get_bboxes(one_stage_head_outs, img_metas)
+        # - two-stage loss
+        roi_losses = self.roi_head.forward_train(bev_feature, img_metas, rois,
+                                                 gt_bboxes_3d, gt_labels_3d)
+        losses.update(roi_losses)
+
+        return losses
+
+    def simple_test(self, points, img_metas, imgs=None, rescale=False):
+        """Test without augmentation.
+
+        Args:
+            points (list[torch.Tensor]): Points of each sample. The outer
+                list indicates point cloud in a batch. The shape of inside
+                point cloud is [N, C]
+            imput_metas (list[dict]): Meta info of each input.
+            imgs (list[torch.Tensor]): Images of each sample.
```

Comment:
imgs (list[torch.Tensor], optional): Images of each sample. Defaults to None.

Reply:
Please also check other places.

Reply:
I mean you need to add the 'default value' and mark it as 'optional' in the docstring.